variables.tf

```diff
@@ -1,5 +1,50 @@
-variable "example" {
-  default     = "default value"
-  type        = "string"
-  description = "Example variable."
+variable "vpc_id" {
+  type        = string
+  description = "The ID of the VPC in which to attach the security group"
+}
+
+variable "sg_name_prefix" {
+  type        = string
+  description = "Prefix for security group names"
+}
+
+variable "ingress_cidr_blocks" {
+  type        = list(string)
+  description = "CIDR blocks to attach to security groups for ingress"
+  default     = []
+}
+
+variable "ingress_security_groups" {
+  type        = list(string)
+  description = "Existing security groups to attach to new security groups for ingress"
+  default     = []
+}
+
+variable "egress_cidr_blocks" {
+  type        = list(string)
+  description = "CIDR blocks to attach to security groups for egress"
+  default     = []
+}
+
+variable "egress_security_groups" {
+  type        = list(string)
+  description = "Existing security groups to attach to new security groups for egress"
+  default     = []
+}
+
+variable "tags" {
+  type        = map(string)
+  description = "Tags to be attached to the resources created"
+  default     = {}
+}
+
+variable "ingress_ports" {
```

Comment:
Each of the services will need a different set of ports here. This could have been an empty list, but it's very likely that this will get set by anyone using this module, so by making it required it's a good reminder to actually set this field.